### ISSUE TITLE
Strange test name, version 3.1.0

### ISSUE THREAD

**Issue by arut-grigoryan** [smarie/python-pytest-cases#171]
Python 3.6.8
pytest-6.1.2
cases-3.1.0

test file
```
@pytest.fixture
def validationOff():
    app.config["validation"] = False
    yield
    app.config["validation"] = True

@parametrize_with_cases("inputs,request_data", prefix="case_create_")
@pytest.mark.usefixtures("clearMongo", "mocked_oid", "mocked_uuid")
def test_create(db, flaskClient, inputs, request_data):
    ...
```
db, flaskClient, clearMongo, mocked_oid, mocked_uuid - fixtures

case file
```
class CreateCases(object):
    @case(id="current_object")
    def case_create_1(self):
        inputs = ["foo", "bar"]
        request_data = {"foo": "dfg", "data": "asd"}
        return inputs, request_data

    @case(id="lala")
    def case_create_10(self):
        inputs = ["foo", "bar"]
        request_data = {"foo": "dfg", "data": "asd"}
        return inputs, request_data
        
    @case(id="without_validation")
    @pytest.mark.usefixtures("validationOff")
    def case_create_2(self):
        inputs = ["foo", "bar"]
        request_data = {"foo": "dfg", "data": "asd"}
        return inputs, request_data

    @case(id="history_object")
    def case_create_3(self):
        inputs = ["foo", "bar"]
        request_data = {"foo": "dfg", "data": "asd"}
        return inputs, request_data

    @case(id="xml_with_namespaces")
    def case_create_4(self):
        inputs = ["foo", "bar"]
        request_data = {"foo": "dfg", "data": "asd"}
        return inputs, request_data

    @case(id="duplicate_name")
    def case_create_6(self):
        inputs = ["foo", "bar"]
        request_data = {"foo": "dfg", "data": "asd"}
        return inputs, request_data
        
    @case(id="duplicate_name_without_valid")
    @pytest.mark.usefixtures("validationOff")
    def case_create_61(self):
       inputs = ["foo", "bar"]
        request_data = {"foo": "dfg", "data": "asd"}
        return inputs, request_data

    @case(id="not_enough_namespaceses")
    def case_create_7(self):
       inputs = ["foo", "bar"]
        request_data = {"foo": "dfg", "data": "asd"}
        return inputs, request_data

    @case(id="invalid_xml")
    def case_create_8(self):
        inputs = ["foo", "bar"]
        request_data = {"foo": "dfg", "data": "asd"}
        return inputs, request_data

    @case(id="invalid_xml_without_valid")
    @pytest.mark.usefixtures("validationOff")
    def case_create_9(self):
        inputs = ["foo", "bar"]
        request_data = {"foo": "dfg", "data": "asd"}
        return inputs, request_data
```

test names:
```
test_create[13h#987__36a721f8Z44526!8*720-current_object]
test_create[13h#987__36a721f8Z44526!8*720-lala]
test_create[without_validation]
test_create[13h#987__36a721f8Z44526!8*721-history_object]
test_create[13h#987__36a721f8Z44526!8*721-xml_with_namespaces]
test_create[13h#987__36a721f8Z44526!8*721-duplicate_name]
test_create[duplicate_name_without_valid]
test_create[13h#987__36a721f8Z44526!8*722-not_enough_namespaceses]
test_create[13h#987__36a721f8Z44526!8*722-invalid_xml]
test_create[invalid_xml_without_valid]
```
**13h#987__36a721f8Z44526!8*720** ??
test names when idstyle="compact":
```
test_create[/P0:2-current_object]
test_create[/P0:2-lala]
test_create[/without_validation]
test_create[/P3:6-history_object]
test_create[/P3:6-xml_with_namespaces]
test_create[/P3:6-duplicate_name]
test_create[/duplicate_name_without_valid]
test_create[/P7:9-not_enough_namespaceses]
test_create[/P7:9-invalid_xml]
test_create[/invalid_xml_without_valid]
```
test names when idstyle="explicit":
```
test_create[(inputs,request_data)/P0:2-current_object]
test_create[(inputs,request_data)/P0:2-lala]
test_create[(inputs,request_data)/without_validation]
test_create[(inputs,request_data)/P3:6-history_object]
test_create[(inputs,request_data)/P3:6-xml_with_namespaces]
test_create[(inputs,request_data)/P3:6-duplicate_name]
test_create[(inputs,request_data)/duplicate_name_without_valid]
test_create[(inputs,request_data)/P7:9-not_enough_namespaceses]
test_create[(inputs,request_data)/P7:9-invalid_xml]
test_create[(inputs,request_data)/invalid_xml_without_valid]
```

**Comment by smarie** [comment]
Dammit. this special id `"13h#987__36a721f8Z44526!8*720"` is my only way to create "no id" ids in pytest>=5.4. It works with pytest 6.2 so I guess this is maybe different with pytest 6.1 ?

**Comment by arut-grigoryan** [comment]
The pytest 6.2.0 and 6.2.1 has the same problem

**Comment by smarie** [comment]
Ah, better then. I'll have a look

**Comment by smarie** [comment]
Found it and fixed in 3.1.1, that should ship soon (when travis has completed)
